core_backend/app/configs/llm_prompts.py

```diff
@@ -132,3 +135,33 @@ def get_supported_languages(cls) -> list[str]:
     'Sorry, no relevant information found.'
 
     Found FAQ: {faq}"""
+
+
+class AlignmentScore(BaseModel):
+    """
+    Alignment score of the user's input.
+    """
+
+    model_config = ConfigDict(strict=True)
+
+    reason: str
+    score: float = Field(ge=0, le=1)
+
+    prompt: ClassVar[
+        str
+    ] = """
+        Using only the CONTEXT provided, reply with a score between 0 and 1 with 0.1
+        increments on how factually and logically consistent the claim provided
+        is with the CONTEXT. A factually consistent claims contains only facts
+        that are entailed in the source document. Check if the `statement` is logically
+        consistent with the CONTEXT. Statements that contain hallucinated facts or
+        those not mentioned in the `context` at all should be heavily penalised.
+        Penalize overly specific statements and omissions. Response as a json object
+        with keys `score` and `reason`. The `score` should be a float between 0 and 1.
+        The `reason` should be a string.
+
+        Example Response: {{"score": 0.5,
+        "reason": "Context does not mention anything about aliens in Ohio."}}
```

Comment:
This might be confusing to the model - we could add an example input and and example response here?

Reply:
I want it to understand the JSON schema and it seems to do that well with this. It has not slipped up yet in my testing. Ok if we leave this as is and if we find it causing issues, we can extend it?